### ISSUE TITLE
Change `StringBuilder.Remove` to J2N appropiate methods

### ISSUE THREAD

**Issue by nikcio** [apache/lucenenet#664]
Originally discussed in #654 

TLDR:
`StringBuilder.Remove` should be replaced with `StringBuilder.Delete` (From J2N)
`StringBuilder.Remove` followed by `StringBuilder.Insert` replaced with `StringBuilder.Replace` (From J2N)

Be aware that the J2N methods uses .NET semantics `startIndex` and length where as the Java source uses `startIndex` and `endIndex` which means the parameters doesn't match 1:1. See #654 for more information.

Note: We should be checking against the Lucene 4.8.0 or 4.8.1 source to be sure that the methods match the original Java code.

Lucene 4.8.0 source can be found here: https://github.com/apache/lucene-solr/tree/releases/lucene-solr/4.8.0/lucene
Lucene 4.8.1 source can be found here: https://github.com/apache/lucene-solr/tree/releases/lucene-solr/4.8.1/lucene
